text/0000-std-rpc-api.md

```diff
@@ -0,0 +1,393 @@
+
+Beyond that, on an internal project, we saw roughly **%80 less boilerplate code** compared to performance RPCs and standard RPCs which is also a huge win.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The multiplayer framework provides 2 main network constructs ([ServerRpc](#serverrpc) and [ClientRpc](#clientrpc)) to execute logic on either server-side or client-side. This concept is often called [Remote Procedure Call (RPC)](https://en.wikipedia.org/wiki/Remote_procedure_call) and has wide adoption across the industry.
+
+## RPC Methods
+
+A framework user (Unity developer) can declare multiple RPCs under a `NetworkBehaviour` and inbound/outbound RPC calls will be replicated as a part of its replication in a network frame.
+
+A method turned into an RPC is no longer a regular method, it will have its  own implications on direct calls and in the network pipeline (see [Execution Table](#execution-table)).
+
+### ServerRpc
+
+A `ServerRpc` can be invoked by a client to be executed on the server.
+
+Developer can declare a `ServerRpc` by marking a method with `[ServerRpc]` attribute and making sure to have `ServerRpc` suffix in the method name.
+
+```cs
+[ServerRpc]
+void PingServerRpc(int framekey) { /* ... */ }
+```
+
+Developer can invoke a `ServerRpc` by making a direct function call with parameters:
+
+```cs
+void Update()
+{
+    if (Input.GetKeyDown(KeyCode.P))
+    {
+        PingServerRpc(Time.frameCount); // Client -> Server
+    }
+}
+```
+
+Marking method with `[ServerRpc]` and putting `ServerRpc` suffix to the method name are required, otherwise it will prompt error messages:
+
+```cs
+// Error: Invalid, missing 'ServerRpc' suffix in the method name
+[ServerRpc]
+void Ping(int framekey) { /* ... */ }
+
+// Error: Invalid, missing [ServerRpc] attribute on the method
+void PingServerRpc(int framekey) { /* ... */ }
+```
+
+`[ServerRpc]` attribute and matching `...ServerRpc` suffix in the method name are there to make it crystal clear for RPC call sites to know when they are executing an RPC, it will be replicated and executed on the server-side, without necessarily jumping into original RPC method declaration to find out if it was an RPC, if so whether it is a ServerRpc or ClientRpc:
+
+```cs
+Ping(framekey); // Is this an RPC call?
+
+PingRpc(framekey); // Is this a ServerRpc call or ClientRpc call?
+
+PingServerRpc(framekey); // This is clearly a ServerRpc call
+```
+
+### ClientRpc
+
+A `ClientRpc` can be invoked by the server to be executed on a client.
+
+Developer can declare a `ClientRpc` by marking a method with `[ClientRpc]` attribute and making sure to have `ClientRpc` suffix in the method name.
+
+```cs
+[ClientRpc]
+void PongClientRpc(int framekey) { /* ... */ }
+```
+
+Developer can invoke a `ClientRpc` by making a direct function call with parameters:
+
+```cs
+void Update()
+{
+    if (Input.GetKeyDown(KeyCode.P))
+    {
+        PongClientRpc(Time.frameCount); // Server -> Client
```

Comment:
What happens when i call PongClientRPC on a Client?

I would understand that it would throw an error, but coming from a Photon PUN background, and being used to relay, it would be very convenient that this would become an automatic-relay-RPC. The server would recieve the message, and immediately relay it to the clients, optionally defined in the ClientSendParams argument.

Reply:
Please see "Execution Table" section.